### ISSUE TITLE
detectBrowserLanguage does not work in production

### ISSUE THREAD

**Issue by J43fura** [nuxt-modules/i18n#2984]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v20.11.0
- Nuxt Version:     3.11.1
- CLI Version:      3.11.1
- Nitro Version:    2.9.4
- Package Manager:  npm@10.5.2
- Builder:          -
- User Config:      srcDir, imports, $development, app, runtimeConfig, modules, primevue, i18n, postcss, css, supabase
- Runtime Modules:  @nuxt/test-utils/module@3.12.0, @nuxtjs/supabase@1.1.7, @nuxtjs/tailwindcss@6.11.4, @pinia/nuxt@0.5.1, nuxt-primevue@0.3.1, @nuxtjs/i18n@8.3.1
- Build Modules:    -
------------------------------

### Reproduction

- The configuration:
```js
  i18n: {
    locales: ['en', 'fr'],
    defaultLocale: 'en',
    strategy: 'no_prefix',
    detectBrowserLanguage: {
      useCookie: true,
      cookieKey: 'i18n_redirected',
      redirectOn: 'root', // recommended
    },
  },
```
- Run `npm run generate`

### Describe the bug

- Locally in development mode (`npm run dev`): The cookie is set correctly with the browser locale being its value, works as expected :heavy_check_mark: 
- Production (`npm run generate`): The cookie's value is set to the defaultLocale instead :x:

**Comment by J43fura** [comment]
reproduction: https://stackblitz.com/edit/github-oyrjnm?file=app.vue

**Comment by J43fura** [comment]
@kazupon here is the reproduction: https://stackblitz.com/edit/github-oyrjnm?file=app.vue

**Comment by codeofsumit** [comment]
I have the same problem - came here trying to debug this and find a solution. With nuxt generate, cookies and browser language are ignored - it's always the default language (and the cookie value is overwritten to the default language on load)

**Comment by BobbieGoede** [comment]
This should be resolved by #3087 which is included in the latest patch release so I'll close this, let me know if you have any questions!

**Comment by codeofsumit** [comment]
I can confirm it works again - thank you for the fix! 🙏
